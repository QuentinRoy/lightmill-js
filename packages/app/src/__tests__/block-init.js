/* globals document */

import simulant from 'simulant';
import initBlock from '../block-init';

let node;
let createView;
beforeEach(() => {
  node = document.createElement('div');
  node.innerHTML = '<div>Foo</div>';
  createView = () =>
    initBlock(node, {
      number: 10,
      measuredBlockNum: 5,
      practice: false,
      factorValues: [
        { factor: { name: 'mock-factor-name' }, name: 'mock-factor-value-name' }
      ]
    });
});

describe('Block view', () => {
  it('can be shown', () => {
    createView();
    expect(node).toMatchSnapshot();
  });

  it('can be removed programmatically', () => {
    createView().remove();
    expect(node).toMatchSnapshot();
  });

  it('is removed when clicked', async () => {
    createView();
    await simulant.fire(node.querySelector('.lightmill-base-app'), 'click');
    expect(node).toMatchSnapshot();
  });

  it('the returned promises resolves when it is removed', async () => {
    let resolved = false;
    const blockViewProm = createView();
    blockViewProm.then(() => {
      resolved = true;
    });

    // Sanity check to make sure the promise does not resolve prematurely.
    await new Promise(resolve => setTimeout(resolve, 0));
    expect(resolved).toBe(false);

    await blockViewProm.remove();
    expect(resolved).toBe(true);
  });

  it('the returned promises resolves when it is clicked', async () => {
    let resolved = false;
    createView().then(() => {
      resolved = true;
    });

    // Sanity check to make sure the promise does not resolve prematurely.
    await new Promise(resolve => setTimeout(resolve, 0));
    expect(resolved).toBe(false);

    await simulant.fire(node.querySelector('.lightmill-base-app'), 'click');
    expect(resolved).toBe(true);
  });
});
