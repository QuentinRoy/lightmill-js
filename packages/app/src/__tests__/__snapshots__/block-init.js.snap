// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block view can be removed programmatically 1`] = `<div />`;

exports[`Block view can be shown 1`] = `
<div>
  <div
    class="lightmill-base-app lightmill-base-app-block"
  >
    <div
      class="lightmill-block"
    >
      <div
        class="lightmill-message"
      >
        Time for a break?
      </div>
      <h1>
        Block NaN
      </h1>
      <div
        class="lightmill-values"
      >
        <div
          class="lightmill-value"
        >
          mock-factor-name: mock-factor-value-name
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Block view is removed when clicked 1`] = `<div />`;
